Rename doctor-key-type-keys to document-field-types (#1163)
papis/papis: doc/source/default_settings.rst

    ``papis --pick-lib export --all``. The format takes a dictionary named
    ``library`` with the keys *name*, *dir*, and *paths*.

.. papis-config:: document-field-types

>    A list of strings ``key:type`` items defining expected types for document

Reviewer: ```suggestion
   A list of ``key:type`` strings defining expected types for document
```

>    keys. The type should be a builtin Python type. For example, this can be

Reviewer: ```suggestion
   fields. The type should be a builtin Python type. For example, this can be
```

   ``["year:int", "tags:list"]`` to check that the year is an integer and the
   tags are given as a list in a document. These values are used in the ``papis
   doctor`` check ``key-type``, in the ``papis update`` command, etc.

   **Note**: This configuration option was previously available as
   ``doctor-key-type-keys`` and ``doctor-key-type-check-keys``. Both of
   these are deprecated (starting with Papis 0.16) and will be removed.

.. papis-config:: document-field-types-extend
    :type: :class:`~typing.List` [:class:`str`]

>     A list of keys that extend the default ones from :confval:`document-field-types`.

Reviewer: ```suggestion
    A list of ``key:type`` strings that extend the default ones from :confval:`document-field-types`.
```

Reviewer: Here and below I'm also happy with 'string items' if you prefer that :)

>     This list extends instead of overwriting the given keys.

Reviewer: ```suggestion
    By setting these, you can extend the defaults rather than overwriting them.
```

   and are just used to add style to the resulting citation (e.g. marking titles
   as italic).

> .. papis-config:: default-key-types

Author: Naming?

Reviewer: maybe `document-schema`? it's a bit more descriptive of what it is

Author: I guess. One worry I have is that we don't actually impose the types anywhere and `document-schema` sounds very "this is what your document will be from now on".

I'm not sure what a good name would be.. Will give it a day or two..

Reviewer: maybe `document-field-types`?

Author: That sounds good to me. It's renamed now :grin: Let me know if anything else looks fishy.
